### PR TITLE
[KVstore] Improve user infterface

### PR DESCRIPTION
**Pull request by aksnzhy**
## Description

1. The client automatically gets metadata (partition-book, data-name, dtype, shape).

```
client:
  name_list = client.get_data_name_list()
  for name in name_list:
      meta = client.get_data_meta(name)
```

2. Add an index to temporary files.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/contrib/dis_kvstore.py [hunk 1]
```diff
@@ -189,23 +190,61 @@ def set_global2local(self, name, global2local=None):
...
-            data_shape = self._read_data_shape(name+'-g2l-shape')
+            data_shape = self._read_data_shape(name+'-g2l-shape-'+str(self._machine_id))
             shared_data = empty_shared_mem(name+'-g2l-', False, data_shape, 'int64')
             dlpack = shared_data.to_dlpack()
             self._data_store[name+'-g2l-'] = F.zerocopy_from_dlpack(dlpack)
 
         self._has_data.add(name+'-g2l-')
 
 
+    def set_partition_book(self, name, partition_book=None):
```
Comment (zheng-da) [inline python/dgl/contrib/dis_kvstore.py:210]: can you move the function back? I'm not sure if I should review this function.

File: python/dgl/contrib/dis_kvstore.py [hunk 2]
```diff
@@ -296,6 +336,40 @@ def get_message_count(self):
         return self._msg_count
 
 
+    def get_data_name_list(self):
```
Comment (zheng-da) [inline python/dgl/contrib/dis_kvstore.py:339]: do we need to define this on the kvstore server?

Comment (aksnzhy) [reply python/dgl/contrib/dis_kvstore.py:339]: As written in our quip doc: All the information should be available in the kvstore server. So I remain this API on server.

File: python/dgl/contrib/dis_kvstore.py (hunk 1 above)
Comment (aksnzhy) [reply python/dgl/contrib/dis_kvstore.py:210]: I just move this function from the client side to the server side.

File: python/dgl/contrib/dis_kvstore.py (hunk 2 above)
Comment (zheng-da) [reply python/dgl/contrib/dis_kvstore.py:339]: I see. By "All the information should be available in the kvstore server", what I mean is that the information is available on the server, so the client can get the information from the server directly. I didn't mean we should provide this API on the server as well.

File: python/dgl/contrib/dis_kvstore.py (hunk 1 above)
Comment (zheng-da) [reply python/dgl/contrib/dis_kvstore.py:210]: I see. My bad. I thought you just move the code around, which makes the review difficult.

**Comment by HuGanghui** [comment]
@aksnzhy  a tiny typo in title: "infterface" -> "interface"

**Pull request merged**